### PR TITLE
Drop SQLite support - test with Postgresql

### PR DESCRIPTION
**Pull request by coolo**
Not finished

### CODE REVIEW

**Comment by coolo** [comment]
@mudler you will need to take over - with postgresql there are no failures, but the full scheduler test expects 8 of them :)

**Comment by mudler** [comment]
🎉  It's all about teamwork!

teamwork

**Comment by okurz** [comment]
How about the still mentioned sqlite references?

```
$ git grep -i sqlite
.gitignore:/t/data/openqa/db/*.sqlite
.gitignore:t/full-stack.d/openqa/db/db.sqlite
Makefile:	install -D -m 640 /dev/null "$(DESTDIR)"/var/lib/openqa/db/db.sqlite
cpanfile:requires 'DBD::SQLite';
dbicdh/SQLite/deploy/37/001-auto-__VERSION.sql:-- Created by SQL::Translator::Producer::SQLite
…
dbicdh/_source/deploy/60/001-auto.yml:    sqlite_version: 3.7
docker/webui/Dockerfile:      /var/lib/openqa/db/db.sqlite /var/lib/openqa/testresults && \
docker/webui/Dockerfile:    ln -s /data/db/db.sqlite /var/lib/openqa/db/db.sqlite && \
docker/webui/run_openqa.sh:if [ ! -f /data/db/db.sqlite ]; then
…
lib/OpenQA/Worker/Cache.pm:    $db_file = catdir($location, 'cache.sqlite');
lib/OpenQA/Worker/Cache.pm:    $dsn = "dbi:SQLite:dbname=$db_file";
t/25-cache.t:my $db_file = "$cachedir/cache.sqlite";
t/25-cache.t:      = DBI->connect("dbi:SQLite:dbname=$db_file", undef, undef, {RaiseError => 1, PrintError => 1, AutoCommit => 1});
t/25-cache.t:ok(-e $db_file, "cache.sqlite is present");
t/full-stack.t:    my $db_file  = $cache_location->child('cache.sqlite');
t/full-stack.t:    ok(!-e $db_file, "cache.sqlite is not present");
t/full-stack.t:    ok(-e $db_file, "cache.sqlite file created");
t/full-stack.t:      = DBI->connect("dbi:SQLite:dbname=$db_file", undef, undef, {RaiseError => 1, PrintError => 1, AutoCommit => 1});
```

**Comment by sysrich** [comment]
Farewell SQLite, you will be mourned.

**Comment by foursixnine** [comment]
Sqlite support is still needed for cache.

**Comment by coolo** [comment]
the dbicdh/SQLite I didn't remove yet to ease the review. If we only support one DB backend, we can simplify that directory a lot actually - to make schema changes reviewable :)

**Comment by coolo** [comment]
to add to @foursixnine's statement: the worker is using sqlite to manage asset cache. The webui is not.

**Comment by coolo** [comment]
FYI: the package builds - the chromium tests still don't work in OBS though, but DB support in general works

**Comment by aaannz** [comment]
Do I understand it right that migration from sqlite db to pgsql is not supported?

**Comment by coolo** [comment]
right. postgresql has always been recommended for production though - and migrating from playground to production was never supported.

The last time we successfully migrated from sqlite to postgresql was with schema 22

File: script/upgradedb [hunk 1]
```diff
@@ -75,26 +75,16 @@ if ($user) {
...
-
-if ($schema->dsn =~ /:SQLite:dbname=(.*)/) {
-    # Some SQLite update scripts do not work correctly with foreign keys on
-    $schema->storage->dbh_do(
-        sub {
-            my ($storage, $dbh, @args) = @_;
-            $dbh->do("PRAGMA foreign_keys = OFF;");
-        });
-}
+my @databases        = qw( PostgreSQL );
```
Comment (asmorodskyi) [inline script/upgradedb]: why you need array for single string ?

Comment (coolo) [reply script/upgradedb]: because strings aren't arrays, hashes are, arrays aren't

**Comment by foursixnine** [comment]
One tip for running full stack tests locally when you don't want to use the postgres user: droping the database and creating it again seems to be more effective,

**Comment by coolo** [comment]
I actually use t/test_postgresql to start a postgresql in tmp running under my user.

**Pull request merged**